src/main/webapp/app/components/commentIcon/CommentIcon.tsx

```diff
@@ -0,0 +1,160 @@
+import React, { useEffect, useMemo, useRef, useState } from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faComment } from '@fortawesome/free-solid-svg-icons/faComment';
+import { Comment } from 'app/shared/model/firebase/firebase.model';
+import { Button, Col, Container, Input, InputGroup, ListGroup, ListGroupItem, Popover, PopoverBody, Row } from 'reactstrap';
+import { SizeProp } from '@fortawesome/fontawesome-svg-core';
+import { CLOSE_POPOVER_DURATION_MILLISECONDS } from 'app/config/constants/constants';
+import './comment-icon.scss';
+
+export interface ICommentIconProps {
+  size: SizeProp;
+  comments: Comment[];
+  onCreateComment: (content: string) => void;
+  onResolveComment: (index: number) => void;
+  onUnresolveComment: (index: number) => void;
+  onDeleteComment: (index: number) => void;
+}
+
+function CommentIcon(props: ICommentIconProps) {
+  let color: string;
+  if (props.comments.length === 0) {
+    color = 'black';
+  } else if (props.comments.some(comment => comment.resolved === 'false')) {
+    color = 'gold';
+  } else {
+    color = 'green';
+  }
```

Comment:
The color should be a computed value so when comments change, the color will be updated.

Reply:
I believe this will automatically update the color? Since the VUS listener is a state, when the comments change, that should trigger a rerender which should rerun this logic.